### PR TITLE
deliver: respect the '--platform' command-line setting

### PR DESCRIPTION
**Pull request by elliot-nelson**
### GOAL
Allow users to successfully upload AppleTV apps to iTunes Connect (using `--platform appletvos`).  See issue https://github.com/fastlane/fastlane/issues/7623.

### CHANGES
* Modify `Deliver::Runner` so that it correctly passes the value of `options[:platform]` (valid values are `ios`, `appletvos`, and `osx`) to the various backend tools.
* Modify `Deliver::DetectValues` to guess the platform correctly when possible (the platform check now comes first, before the other method calls which look at `options[:platform]`).
* Add some basic tests for `Deliver::Runner#upload_binary`.

### TESTS
- [x] Run `bundle exec rspec` from the root directory.
- [x] Run `bundle exec rubocop -a` to ensure the code style is valid
- [x] Read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] We currently don't accept new actions, please publish a plugin instead, more information in [Plugins.md](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Plugins.md)

### CODE REVIEW

File: deliver/spec/runner_spec.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      it 'uploads the IPA for the appletvos platform' do
+        expect_any_instance_of(FastlaneCore::IpaUploadPackageBuilder).to receive(:generate)
+          .with(app_id: 'YI8C2AS', ipa_path: 'ACME.ipa', package_path: '/tmp', platform: 'appletvos')
+          .and_return('path')
+        expect(transporter).to receive(:upload).with('YI8C2AS', 'path').and_return(true)
+        runner.upload_binary
+      end
+    end
+
+    describe 'with a PKG file for Mac OSX' do
```
Comment (JustinJiaDev) [inline deliver/spec/runner_spec.rb]: `with a PKG file for macOS`?

File: deliver/spec/runner_spec.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      it 'uploads the IPA for the iOS platform' do
+        expect_any_instance_of(FastlaneCore::IpaUploadPackageBuilder).to receive(:generate)
+          .with(app_id: 'YI8C2AS', ipa_path: 'ACME.ipa', package_path: '/tmp', platform: 'ios')
+          .and_return('path')
+        expect(transporter).to receive(:upload).with('YI8C2AS', 'path').and_return(true)
+        runner.upload_binary
+      end
+    end
+
+    describe 'with an IPA file for AppleTV' do
```
Comment (JustinJiaDev) [inline deliver/spec/runner_spec.rb]: `with an IPA file for tvOS`?

File: deliver/spec/runner_spec.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    end
+
+    describe 'with a PKG file for Mac OSX' do
+      before do
+        options[:platform] = 'osx'
+        options[:pkg] = 'ACME.pkg'
+        options[:ipa] = nil
+      end
+
+      it 'uploads the PKG for the osx platform' do
```
Comment (JustinJiaDev) [inline deliver/spec/runner_spec.rb]: `uploads the PKG for the macOS platform`?

**Comment by elliot-nelson** [comment]
^ Feedback incorporated (more consistent descriptions in new specs).

**Comment by KrauseFx** [comment]
Hey @elliot-nelson, thanks for your pull request, this looks great. I'd love to get a review from @tmm1 before merging, as he built most of the tvOS support in _fastlane_ 👍

**Comment by tmm1** [comment]
LGTM

**Review by asfalcone — APPROVED** [review]

**Comment by asfalcone** [comment]
Thanks! 🎉

**Pull request merged**